Correctly handle query strings that contain `=`

genomics-geek opened the pull request:
Some of our GraphQL IDs had `=` in their value, therefore splitting on `=` creates issues - unless we split by the 1st occurrence.

For example: `?testCode_In=VGVzdENvZGVOb2RlOjQ=` will be parsed as: `{testCode_In: "VGVzdENvZGVOb2RlOjQ"}`.  This leads to confusion and inaccurate search terms.

This has resolved the issue for us.

sindresorhus commented on index.js:
```diff
@@ -150,7 +150,7 @@ function parse(input, options) {
 
 	for (const param of input.split('&')) {
-		let [key, value] = param.replace(/\+/g, ' ').split('=');
+		let [key, value] = param.replace(/\+/g, ' ').split(/\=(.+)/); // eslint-disable-line no-useless-escape
```
ESLint is correct. You don't have to escape `=`.

genomics-geek replied:
@sindresorhus - yup I know, but then I get a different ESLint error about escaping.   Yup I will add a test.  Also, it seems to fail other tests now, so I have to look into it

sindresorhus replied:
Ah ok. Better to ignore the `no-div-regex` rule.

sindresorhus commented:
Can you add a test?

genomics-geek commented:
I added tests and replaced the regex with a `replace().split()` - not as nice as regex, but it avoids adding extra element to the split.  Let me know what you think @sindresorhus

cam-perry commented on index.js:
```diff
@@ -150,7 +150,7 @@ function parse(input, options) {
 
 	for (const param of input.split('&')) {
-		let [key, value] = param.replace(/\+/g, ' ').split('=');
+		let [key, value] = param.replace(/\+/g, ' ').replace('=', '<**>').split('<**>');
```
Just ran into the same issue. This is the alternative I used for a fix https://github.com/cam-perry/query-string/commit/b65a53b55ae7a8825d62af757a587d365ba7e3b7:
```js
const i = param.indexOf("=");
let key = param.slice(0, i);
let value = param.slice(i+1);
```

sindresorhus replied:
The previous regex was better. This is risky as it will fail if the query string actually contains `<**>`, even if unlikely.

MatheusFelipe commented:
We're also experiencing this issue. When will the fix be merged?

sindresorhus commented:
You can use my https://github.com/sindresorhus/split-on-first module.

genomics-geek commented:
@sindresorhus - Nice.  I used your package like you suggested.  This should take care of the GraphQL ids now 👍

genomics-geek commented:
@sindresorhus - is there anything else that is needed in order to merge?
